js/source/tile_coord.js

```diff
@@ -51,12 +51,12 @@ TileCoord.fromID = function(id) {
 };
 
 // given a list of urls, choose a url template and return a tile URL
-TileCoord.prototype.url = function(urls, sourceMaxZoom) {
+TileCoord.prototype.url = function(urls, sourceMaxZoom, scheme) {
     return urls[(this.x + this.y) % urls.length]
         .replace('{prefix}', (this.x % 16).toString(16) + (this.y % 16).toString(16))
         .replace('{z}', Math.min(this.z, sourceMaxZoom || this.z))
         .replace('{x}', this.x)
-        .replace('{y}', this.y);
+        .replace('{y}', scheme === 'tms' ? (Math.pow(2, this.z) - this.y - 1) : this.y);
```

Comment:
Hi, I'm using mbtiles, but i find the tool i used to translate google xyz to tms schema does not use `Math.pow(2, this.z) - this.y - 1`, but `Math.pow(2, this.z - 1) - this.y - 1`. I konw that tms's original is (-180, -90), and xyz's original is (-180, 90), so it is easy to understand `Math.pow(2, this.z - 1) - this.y - 1`. 

Can somebody tell me how does `Math.pow(2, this.z) - this.y - 1` works and which is right method?

Reply:
You may notice, that `vertor_zoom = raster_zoom - 1`.

Reply:
Can you provide a link to the tool you are using?
With quick websearch I only found the conversation in use:
https://gist.github.com/tmcw/4954720
http://bl.ocks.org/lennepkade/b6fe9e4862668b2d19fe26f6c2d7cbef
https://alastaira.wordpress.com/2011/07/06/converting-tms-tile-coordinates-to-googlebingosm-tile-coordinates/

Reply:
I've red the 3rd blog and learned that i misunderstood the map tile method. So we transfer the world to a square using Web Mercator system, and we can get the total tile number which can be described as (1 << zoom - 1).
Now I guess it is the coordinate system i'm using cause the problem. In fact, i transformed a map to mbtiles which is projected to `EPSG:4326` (WGS84) . But as we know mbtiles spec using tms and mercator projection.  That might need to be supported by a projection-customized mapbox-gl.
Thanks for your help! :)  :+1: